core/src/tasks/pick_place_task.cpp

```diff
@@ -131,254 +131,54 @@ void PickPlaceTask::init(const Parameters& parameters)
-      grasp->insert(std::move(stage));
-    }
-
-    /****************************************************
-  .... *               Attach Object                      *
-     ***************************************************/
-    {
-      auto stage = std::make_unique<stages::ModifyPlanningScene>("attach object");
-      stage->attachObject(parameters.object_name_, parameters.hand_frame_);
-      attach_object_stage = stage.get();
-      grasp->insert(std::move(stage));
-    }
-
-    /****************************************************
-  .... *               Allow collision (object support)   *
-     ***************************************************/
-    {
-      auto stage = std::make_unique<stages::ModifyPlanningScene>("allow collision (object,support)");
-      stage->allowCollisions({ parameters.object_name_ }, "source_container", true);  // TODO(henningkayser): parameterize
-      grasp->insert(std::move(stage));
-    }
-
-    /****************************************************
-  .... *               Lift object                        *
-     ***************************************************/
-    {
-      auto stage = std::make_unique<stages::MoveRelative>("lift object", cartesian_planner);
-      stage->properties().configureInitFrom(Stage::PARENT, { "group" });
-      stage->setMinMaxDistance(parameters.lift_object_min_dist_, parameters.lift_object_max_dist_);
-      stage->setIKFrame(parameters.hand_frame_);
-      stage->properties().set("marker_ns", "lift_object");
-
-      // Set upward direction
-      geometry_msgs::Vector3Stamped vec;
-      vec.header.frame_id = parameters.world_frame_;
-      vec.vector.z = 1.0;
-      stage->setDirection(vec);
-      grasp->insert(std::move(stage));
-    }
-
-    /****************************************************
-  .... *               Forbid collision (object support)  *
-     ***************************************************/
-    {
-      auto stage = std::make_unique<stages::ModifyPlanningScene>("forbid collision (object,surface)");
-      stage->allowCollisions({ parameters.object_name_ }, "source_container", false);  // TODO(henningkayser): parameterize
-      grasp->insert(std::move(stage));
-    }
-
-    // Add grasp container to task
-    t.add(std::move(grasp));
-  }
-
-  /******************************************************
-   *                                                    *
-   *          Move to Place                             *
-   *                                                    *
-   *****************************************************/
-  {
-    auto stage = std::make_unique<stages::Connect>(
-        "move to place", stages::Connect::GroupPlannerVector{ { parameters.arm_group_name_, sampling_planner } });
-    stage->setTimeout(5.0);
-    stage->properties().configureInitFrom(Stage::PARENT);
+    auto stage = std::make_unique<stages::Pick>("Pick object", parameters.grasp_provider_plugin_name_);
```

Comment:
This seems a lot more difficult to read now. I also don't know that this example code is the right place to explain how to use the capability (I assume that that's what you're trying to do). The MTC example code would still demonstrate the MTC internals and how to make new tasks. End users who just want to use the PickPlace capability with the `move_group_interface` would look at the MoveIt tutorial. I would update the `PickPlace` tutorial there instead.

Reply:
Oh, I think I might have misunderstood. This is actually used in the capability, right? Hmmm.